pulp_rpm/tests/functional/api/test_crud_content_unit.py

```diff
@@ -158,3 +171,64 @@ def test_06_second_unit_raises_error(self):
         )
 
         self.assertEqual(task["error"]["description"], error_message)
+
+
+class ContentUnitRemoveTestCase(PulpTestCase):
+    """
+    Test of content removal.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class-wide variable."""
+        delete_orphans()
+        cls.client = gen_rpm_client()
+        cls.cfg = Configuration(cls.client)
+        cls.repo_api = RepositoriesRpmApi(cls.client)
+        cls.remote_api = RemotesRpmApi(cls.client)
+
+    def do_test_remove_unit(self, remote_url):
+        """
+        Sync repository and test that content can't be removed directly.
+        """
+        repo = self.repo_api.create(gen_repo())
+        remote_body = gen_rpm_remote(remote_url)
+        remote = self.remote_api.create(remote_body)
+        repository_sync_data = RpmRepositorySyncURL(remote=remote.pulp_href)
+        sync_response = self.repo_api.sync(repo.pulp_href, repository_sync_data)
+        monitor_task(sync_response.task)
+        repo = self.repo_api.read(repo.pulp_href)
+
+        # add resources to clean up
+        self.addCleanup(self.repo_api.delete, repo.pulp_href)
+        self.addCleanup(self.remote_api.delete, remote.pulp_href)
+
+        # Test remove content by types contained in repository.
+        repo_content = get_content(repo.to_dict())
+        base_addr = self.cfg.get_host_settings()[0]['url']
+
+        for content_type in repo_content.keys():
+            response = requests.delete(
+                urljoin(base_addr, repo_content[content_type][0]["pulp_href"])
+            )
+            # check that '405' (method not allowed) is returned
+            self.assertEqual(
+                response.status_code, 405
+            )
+
+    def test_all(self):
+        """
+        Test two repositories to cover mony content types.
+
+        - advisory
+        - distribution_tree
+        - modulemd
+        - modulemd_defaults
+        - package
+        - packagecategory
+        - packageenvironment
+        - packagegroup
+        - packagelangpacks
+        """
+        self.do_test_remove_unit(RPM_MODULAR_FIXTURE_URL)
```

Comment:
i see custom repodata is missing here, can you add 3a rd repo that will test its removal? this way we will have covered all rpm content types.